ntp-daemon/src/peer.rs

```diff
@@ -137,27 +135,30 @@ pub async fn start_peer<A: ToSocketAddrs + std::fmt::Debug, C: 'static + NtpCloc
                                 recv_timestamp,
                             );
 
-                            let system_poll = system_snapshot.poll_interval.as_duration();
-                            let accept = peer.accept_synchronization(
-                                ntp_instant,
-                                config.frequency_tolerance,
-                                config.distance_threshold,
-                                system_poll,
-                            );
-
-                            if accept.is_err() {
-                                let _ = tx.send(MsgForSystem::NoMeasurement);
-                            } else  {
-                                match result {
-                                    Ok(update) => {
```

Comment:
the diff here is not great. what I'm doing is inverting the control flow: we now check the `result` before checking whether we should `accept_synchronization`. In particular that means the DENY kiss code is now correctly propagated